yaksh/evaluator_tests/test_simple_question_types.py

```diff
@@ -435,18 +466,278 @@ def test_float_incorrect_answer(self):
+        regrade_answer = Answer.objects.get(id=self.answer.id)
+
+        # Try regrade with wrong data structure
+        # When
+        regrade_answer.answer = 1
+        regrade_answer.save()
+        details = self.answerpaper.regrade(self.question1.id)
+        err_msg = dedent("""\
+                          User: {0}; Quiz: {1}; Question: {2}.
+                          {3} answer not a list.""".format(
+                                                     self.user.username,
+                                                     self.quiz.description,
+                                                     self.question1.summary,
+                                                     self.question1.type
+                        )                                )
+        self.assertFalse(details[0])
+        self.assertEqual(details[1], err_msg)
+
+
+        # Try regrade with incorrect answer
+        # When
+        regrade_answer.answer = [self.testcase_1_id,
+                                 self.testcase_3_id,
+                                 self.testcase_2_id,
+                                 ]
+        regrade_answer.save()
+        # Then
+        details = self.answerpaper.regrade(self.question1.id)
+        self.answer = self.answerpaper.answers.filter(question=self.question1
+                                                      ).last()
+        self.assertEqual(self.answer, regrade_answer)
+        self.assertTrue(details[0])
+        self.assertEqual(self.answer.marks, 0)
+        self.assertFalse(self.answer.correct)
+
+    def test_validate_regrade_arrange_incorrect_answer(self):
+        # Given
+        arrange_answer = [self.testcase_1_id,
+                          self.testcase_3_id,
+                          self.testcase_2_id,
+                          ]
+        self.answer = Answer(question=self.question1,
+                             answer=arrange_answer,
+                             )
+        self.answer.save()
+        self.answerpaper.answers.add(self.answer)
+
+        # When
+        json_data = None
+        result = self.answerpaper.validate_answer(arrange_answer,
+                                                  self.question1, json_data
+                                                  )
+
+        # Then
+        self.assertFalse(result['success'])
+        # Regrade with wrong answer
+        # Given
+        regrade_answer = Answer.objects.get(id=self.answer.id)
+        regrade_answer.answer = [self.testcase_1_id,
+                                 self.testcase_2_id,
+                                 self.testcase_3_id,
+                                 ]
+        regrade_answer.save()
 
         # When
         details = self.answerpaper.regrade(self.question1.id)
 
         # Then
         self.answer = self.answerpaper.answers.filter(question=self.question1
                                                       ).last()
+        self.assertEqual(self.answer, regrade_answer)
```

Comment:
Yes, I too wondered doing that. But the problem is, since answerpaper is modified in pretty much every test class, I decided against doing that. I believe creating and deleting frequently used objects then and there is more preventive.